source/distributed/shared_tria.cc

```diff
@@ -274,6 +290,83 @@ namespace parallel
       this->update_number_cache ();
     }
 
+
+
+    template <int dim, int spacedim>
+    void
+    Triangulation<dim,spacedim>::
+    update_number_cache ()
+    {
+      parallel::Triangulation<dim,spacedim>::update_number_cache();
```

Comment:
this is the same implementation as p::d::Tria, right? Just move it into p::Tria and delete this and the function update_number_cache() in p::d::Tria.

Reply:
No. We can not move the function to the base class, as it doesn't know about ``settings & construct_multigrid_hierarchy`` (the settings are independently defined in both derived classes). He did move the duplicated code into the new ``fill_level_ghost_owners``.

Reply:
Ah, ok. Thanks for confirming.